packages/ros2/src/SystemHandle.cpp

```diff
@@ -25,6 +25,17 @@
 
 #include <rclcpp/executors/single_threaded_executor.hpp>
 
+#ifdef WIN32
+#define SETENV(id,value,b,retValue) \
```

Comment:
Multi-line macros can get kind of gross very quickly. Can we consider doing something more like:

```
int set_platform_env(const std::string& variable, const std::string& value, const bool overwrite)
{
#ifdef WIN32
  std::ostringstream aux_d;
  aux_d << variable << "=" << value;
  return _putenv(aux_d.str().c_str());
#else
  return setenv(variable, value, overwrite);
#endif // WIN32
}
```

And then similarly for an `unset_platform_env(~)` function.

I know it would be more lines of code, but I think it would help make the code more clear.

Reply:
I completely agree. I have rewrote that in functions.